src/providers/FusionAuthProvider.tsx

```diff
@@ -62,25 +82,14 @@ enum FunctionType {
     register = 'register',
 }
 
-async function generateUrl(
+function generateUrl(
     functionType: FunctionType,
     baseUrl: string,
-    clientID: string,
-    scope: string,
-    redirectUri: string,
-    state = '',
+    queryString: Record<string, string>,
```

Comment:
tThis would be better named `queryParams` because it is not a string, it is an object containing the parameters